base_export_manager/models/ir_exports_line.py

```diff
@@ -53,104 +62,106 @@ def _default_model1_id(self):
+                          if parts and self.name else False)
+        except KeyError:
+            pass
 
-    @api.multi
+    @api.one
     def _inverse_name(self):
         """Get the fields from the name."""
-        for s in self:
-            # Field names can have up to only 3 indentation levels
-            parts = s.name.split("/", 2)
-
-            for num in range(1, 4):
-                try:
-                    # Fail in excessive subfield level
-                    field_name = parts[num - 1]
-                except IndexError:
-                    # Remove subfield on failure
-                    s[s.field_n(num, True)] = False
-                else:
-                    model = s.model_n(num)
-                    s[s.field_n(num, True)] = self._get_field_id(
-                        model, field_name)
+        # Field names can have up to only 3 indentation levels
+        parts = self.name.split("/")
+        if len(parts) > 4:
+            raise exceptions.ValidationError(
+                _("It's not allowed to have more than 4 levels depth: "
+                  "%s") % self.name)
+        for num in range(1, 5):
+            if num > len(parts):
+                # Empty subfield in this case
+                self[self.field_n(num, True)] = False
+                continue
+            field_name = parts[num - 1]
+            model = self.model_n(num)
+            self.with_context(skip_check=True)[self.field_n(num, True)] = (
+                self._get_field_id(model, field_name))
+        self._check_name()
 
     @api.one
-    @api.constrains("field1_id", "field2_id", "field3_id")
+    @api.constrains("field1_id", "field2_id", "field3_id", "field4_id")
     def _check_name(self):
         if not self.label:
             raise exceptions.ValidationError(
                 _("Field '%s' does not exist") % self.name)
-        lines = self.search([('export_id', '=', self.export_id.id),
-                             ('name', '=', self.name)])
-        if len(lines) > 1:
-            raise exceptions.ValidationError(
-                _("Field '%s' already exists") % self.name)
-
-    @api.model
-    def _install_base_export_manager(self):
-        """Populate ``field*_id`` fields."""
-        self.search([("export_id", "=", False)]).unlink()
-        lines = self.search(
-            [("field1_id", "=", False), ("name", "!=", False)])
-        for line in lines:
-            try:
-                line._inverse_name()
-            except:
-                # Prevent possible inexisting fields
-                pass
+        if not self.env.context.get('skip_check'):
+            lines = self.search([('export_id', '=', self.export_id.id),
+                                 ('name', '=', self.name)])
+            if len(lines) > 1:
+                raise exceptions.ValidationError(
+                    _("Field '%s' already exists") % self.name)
+
+    @api.onchange('name')
```

Comment:
`@api.multi` is implicit if the method signature does not start by `self, cr, uid, ...`. Its advantage is that makes methods callable from old api too, but if that's not a requirement, you don't need to put it. I'm more in the side of always decorating, but certainly it's not required.